### ISSUE TITLE
thumbnail path with hash filename

### ISSUE THREAD

**Issue by cutemonster** [Studio-42/elFinder#1515]
Hi,
Is there a way to get the hashed thumbnail file name based on selected image?
I'm currently just displaying the original image on the form preview but I would like to display the thumbnail version instead.  If I know how it's hashed, then I can display the thumbnail version of the image instead.

Thanks!

**Comment by nao-pon** [comment]
@cutemonster It obtained the file.tmb in `getFileCallback`.

```javascript
getFileCallback : function(file, fm) {
	console.log(file.tmb);
}
```

OR If you want to control the naming convention, Please use your custom volume driver that overridden `elFinderVolumeDriver::tmbname()`.

e.g.
```php
class elFinderVolumeMyLocalFileSystem extends elFinderVolumeMyLocalFileSystem
{
	/**
	 * Return thumbnail file name for required file
	 *
	 * @param  array  $stat  file stat
	 * @return string
	 * @author Dmitry (dio) Levashov
	 **/
	protected function tmbname($stat) {
		return $stat['name'].'.png';
	}
}
```

**Comment by cutemonster** [comment]
Hi Nao-pon,
Thanks for the reply.  I extended **elFinderVolumeLocalFileSystem** but I couldn't find any file that use this class.   Could you tell me what I need to do after I extend it?

I'm trying not to modify the original class.

thanks!

**Comment by nao-pon** [comment]
Use `MyLocalFileSystem` as volume driver class. (`'driver' => 'MyLocalFileSystem'`)

Simple example...
```php
<?php

// elFinder autoload
require './autoload.php';

class elFinderVolumeMyLocalFileSystem extends elFinderVolumeLocalFileSystem
{
    /**
     * Return thumbnail file name for required file
     *
     * @param  array  $stat  file stat
     * @return string
     * @author Dmitry (dio) Levashov
     **/
    protected function tmbname($stat) {
        return $stat['name'].'.png';
    }
}

/**
 * Simple function to demonstrate how to control file access using "accessControl" callback.
 * This method will disable accessing files/folders starting from '.' (dot)
 *
 * @param  string  $attr  attribute name (read|write|locked|hidden)
 * @param  string  $path  file path relative to volume root directory started with directory separator
 * @return bool|null
 **/
function access($attr, $path, $data, $volume) {
	return strpos(basename($path), '.') === 0       // if file/folder begins with '.' (dot)
		? !($attr == 'read' || $attr == 'write')    // set read+write to false, other (locked+hidden) set to true
		:  null;                                    // else elFinder decide it itself
}

// Documentation for connector options:
// https://github.com/Studio-42/elFinder/wiki/Connector-configuration-options
$opts = array(
	// 'debug' => true,
	'roots' => array(
		array(
			'driver'        => 'MyLocalFileSystem',         // driver for accessing file system (REQUIRED)
			'path'          => '../files/',                 // path to files (REQUIRED)
			'URL'           => dirname($_SERVER['PHP_SELF']) . '/../files/', // URL to files (REQUIRED)
			'uploadDeny'    => array('all'),                // All Mimetypes not allowed to upload
			'uploadAllow'   => array('image', 'text/plain'),// Mimetype `image` and `text/plain` allowed to upload
			'uploadOrder'   => array('deny', 'allow'),      // allowed Mimetype `image` and `text/plain` only
			'accessControl' => 'access'                     // disable and hide dot starting files (OPTIONAL)
		)
	)
);

// run elFinder
$connector = new elFinderConnector(new elFinder($opts));
$connector->run();
```

**Comment by cutemonster** [comment]
ah, it's in the autoload file, that's why I couldn't find LocalFileSystem.  Let me try again.
Thanks for the help!!!
